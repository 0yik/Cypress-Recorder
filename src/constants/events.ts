export default {
  CLICK: 'click',
  CHANGE: 'change',
  DBCLICK: 'dbclick',
  KEYDOWN: 'keydown',
  RESET: 'reset',
  SUBMIT: 'SUBMIT',
};
