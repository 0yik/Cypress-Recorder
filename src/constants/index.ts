export enum EventTypes {
  CLICK = 'click',
  CHANGE = 'change',
  DBCLICK = 'dbclick',
  KEYDOWN = 'keydown',
  RESET = 'reset',
  SUBMIT = 'submit',
}
