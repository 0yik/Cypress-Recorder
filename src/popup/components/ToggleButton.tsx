import * as React from 'react';
import { ControlAction } from '../../constants';

export interface ToggleButtonProps {
  recStatus: String,
  handleToggle: (action: ControlAction) => void,
}

export default ({ recStatus, handleToggle }: ToggleButtonProps) => {
  const handleClick = (): void => {
    let action: ControlAction | null = null;
    if (recStatus === 'off') action = ControlAction.START;
    else if (recStatus === 'on') action = ControlAction.STOP;
    else if (recStatus === 'done') action = ControlAction.RESET;
    if (action) {
      handleToggle(action);
    }
  };

  return (
    <>
      <button className="button" onClick={handleClick}>
        {recStatus === 'off'
          ? 'Start Recording'
          : recStatus === 'on'
            ? 'Stop Recording'
            : 'Reset Recording'}
      </button>
    </>
  );
}
