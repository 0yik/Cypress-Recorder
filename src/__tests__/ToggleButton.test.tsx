import * as React from 'react';
import { shallow } from 'enzyme';
import ToggleButton from '../popup/components/ToggleButton';
import { ControlAction } from '../constants';
import '../setupTests';

describe('toggle button', () => {
  const handleToggleMock = jest.fn();
  let wrapper;
  let props;
  beforeAll(() => {
    props = {
      handleToggle: handleToggleMock,
    };
  });
  it('Should call handleToggle with argument "startRec" when clicked and recStatus is "off" and tab is valid', () => {
    props.recStatus = 'off';
    props.isValidTab = true;
    wrapper = shallow(<ToggleButton { ...props } />);
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    expect(button.text()).toBe('Start Recording');
    button.simulate('click');
    expect(handleToggleMock).toHaveBeenCalledWith(ControlAction.START);
  });
  it('Should call handleToggle with argument "stopRec" when clicked and recStatus is "on"', () => {
    props.recStatus = 'on';
    props.isValidTab = true;
    wrapper = shallow(<ToggleButton { ...props } />);
    const button = wrapper.find('button');
    expect(button.text()).toBe('Stop Recording');
    button.simulate('click');
    expect(handleToggleMock).toHaveBeenCalledWith(ControlAction.STOP);
  });
  it('Should call handleToggle with argument "startRec" when clicked and recStatus is "paused"', () => {
    props.recStatus = 'paused';
    props.isValidTab = true;
    wrapper = shallow(<ToggleButton { ...props } />);
    const button = wrapper.find('button');
    expect(button.text()).toBe('Resume');
    button.simulate('click');
    expect(handleToggleMock).toHaveBeenCalledWith(ControlAction.START);
  });
  it('Should be disabled when clicked and recStatus is "off" and tab is invalid', () => {
    props.recStatus = 'off';
    props.isValidTab = false;
    wrapper = shallow(<ToggleButton { ...props } />);
    const button = wrapper.find('button');
    expect(button.text()).toBe('Invalid Tab');
    expect(button.prop('disabled')).toBe(true);
  });
});
