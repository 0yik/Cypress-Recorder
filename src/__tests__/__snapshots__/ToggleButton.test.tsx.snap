// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle button Should call handleToggle with argument "startRec" when clicked and recStatus is "off" and tab is valid 1`] = `
<div
  id="toggle-wrap"
>
  <button
    className="button"
    disabled={false}
    id="toggle"
    onClick={[Function]}
    type="button"
  >
    Start Recording
  </button>
</div>
`;
